addons/knowledge/static/src/js/knowledge_frontend.js

```diff
@@ -0,0 +1,137 @@
+odoo.define('knowledge.knowledge_frontend', function (require) {
+'use strict';
+
+var publicWidget = require('web.public.widget');
+var core = require('web.core');
+
+var QWeb = core.qweb;
+
+publicWidget.registry.KnowledgeWidget = publicWidget.Widget.extend({
+    selector: '.o_knowledge_form_view',
+    events: {
```

Comment:
Global remarks here as well:

- Methods should be correctly documented (ideally with their `@tags` (param, private, override, ...)
- Methods should be split into sections: "Handlers" and "Private" accordingly (you can grep in the code to see how we mark sections with comments)
- Methods should be sorted alphabetically within their section